app/auth/middleware.js

```diff
@@ -13,10 +16,11 @@ function processAuthResponse() {
     try {
       const accessToken = grant.response.access_token
       const decodedAccessToken = decodeAccessToken(grant.response.access_token)
-      const [locations, fullname] = await Promise.all([
-        getLocations(accessToken),
+      const [groups, fullname] = await Promise.all([
+        getGroups(accessToken),
         getFullname(accessToken),
       ])
+      const locations = await getLocationsByNomisAgencyId(groups)
```

Comment:
this feels a little awkward but I cant really see a better way of doing it.

Maybe removing the `Promise.all` and doing 3 awaits?

```
const groups = await getGroups(accessToken)
const fullname =  await getFullname(accessToken)
const locations = await getLocationsByNomisAgencyId(groups)
```